pkg/azureutils/azure_disk_utils.go

```diff
@@ -310,3 +328,95 @@ func checkDiskName(diskName string) bool {
 
 	return true
 }
+
+func GetMaxSharesAndMaxMountReplicaCount(parameters map[string]string) (int, int) {
+	maxShares := 1
+	maxMountReplicaCount := 0
+	for param, value := range parameters {
+		if strings.EqualFold(param, MaxSharesField) {
+			parsed, err := strconv.Atoi(value)
+			if err != nil {
+				klog.Warningf("failed to parse maxShares value (%s) to int, defaulting to 1: %v", value, err)
+			} else {
+				maxShares = parsed
+			}
+		} else if strings.EqualFold(param, MaxMountReplicaCountField) {
+			parsed, err := strconv.Atoi(value)
+			if err != nil {
+				klog.Warningf("failed to parse maxMountReplica value (%s) to int, defaulting to 0: %v", value, err)
+			} else {
+				maxMountReplicaCount = parsed
+			}
+		}
+	}
+
+	if maxShares-1 < maxMountReplicaCount {
+		klog.Warningf("maxMountReplicaCount cannot be larger than maxShares - 1: defaulting to maxShares - 1")
+		maxMountReplicaCount = maxShares - 1
+	}
+	return maxShares, maxMountReplicaCount
+}
+
+func GetAzVolumePhase(phase v1.PersistentVolumePhase) v1alpha1.AzVolumePhase {
+	return v1alpha1.AzVolumePhase(phase)
+}
+
+func GetAzVolume(ctx context.Context, client client.Client, azDiskClient azDiskClientSet.Interface, azVolumeName, namespace string, useCache bool) (*v1alpha1.AzVolume, error) {
```

Comment:
Rather than use a `bool` parameter to choose cached vs. uncache, consider creating a separate `GetXXXUncached` function for each object. Using a purpose-named function makes the behavior much clearer at the call-point than using a value like `true` or `false`.

Reply:
How about we create a type to pass in as a parameter as
```
type OperationMode int
const (
   Cached OperationMode = iota
   Uncached OperationMode
```
Wouldn't this give us clarity for call-point while allowing us to stay away from making two separate get functions for each object?

Reply:
I will push the relevant changes to the refactoring PR once we finalize our go on this :)

Reply:
I am fine with the `OperationMode` parameter approach.